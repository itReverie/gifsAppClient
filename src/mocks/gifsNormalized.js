const gifsNormalized={ "B0vFTrb0ZGDf2": { id: "B0vFTrb0ZGDf2",
                                          isFavorite:true,
                                          preview :{
                                                url: "https://media1.giphy.com/media/B0vFTrb0ZGDf2/200w.gif?cid=4647a4e05b37e8f8747833465127ae26",
                                                width: "200",
                                                height: "105",
                                                size: "74046",
                                                mp4: "https://media1.giphy.com/media/B0vFTrb0ZGDf2/200w.mp4?cid=4647a4e05b37e8f8747833465127ae26",
                                                mp4_size: "28128",
                                                webp: "https://media1.giphy.com/media/B0vFTrb0ZGDf2/200w.webp?cid=4647a4e05b37e8f8747833465127ae26",
                                                webp_size: "146398"}
                                  },
                     "oDLDbBgf0dkis" : { id: "oDLDbBgf0dkis",
                                        isFavorite:true,
                                        preview: {
                                                    url: "https://media3.giphy.com/media/oDLDbBgf0dkis/200w.gif?cid=4647a4e05b37e8f8747833465127ae26",
                                                    width: "200",
                                                    height: "113",
                                                    size: "534324",
                                                    mp4: "https://media3.giphy.com/media/oDLDbBgf0dkis/200w.mp4?cid=4647a4e05b37e8f8747833465127ae26",
                                                    mp4_size: "85697",
                                                    webp: "https://media3.giphy.com/media/oDLDbBgf0dkis/200w.webp?cid=4647a4e05b37e8f8747833465127ae26",
                                                    webp_size: "53300"}
                                }   
};

export default gifsNormalized;

