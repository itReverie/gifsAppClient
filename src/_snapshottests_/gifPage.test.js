import React from 'react';
import GifPage from '../pages/gifPage';
import renderer from 'react-test-renderer';
import gifsNormalized from '../mocks/gifsNormalized';
import {BrowserRouter} from 'react-router-dom';


it('rendering correctly a Gifs page via snapshot', () => {
  const onAddFavorite=jest.fn();
  const onSearch=jest.fn();
  const onUpdateError=jest.fn();

  const tree = renderer
    .create(<BrowserRouter><GifPage gifslist={gifsNormalized}
                     onAddFavorite={onAddFavorite}
                     onSearch={onSearch}
                     error={{}}
                     onUpdateError={onUpdateError} /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});