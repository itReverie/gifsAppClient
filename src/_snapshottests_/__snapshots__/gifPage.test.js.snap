// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering correctly a Gifs page via snapshot 1`] = `
<div
  className="sc-fjdhpX koYQtA"
>
  <div
    className="sc-iwsKbI gVQksL"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        className="sc-gZMcBi eEyyxI"
        src="home.svg"
      />
    </a>
    <div
      className="sc-VigVT dBjWVv"
      onClick={[Function]}
    >
      <img
        className="sc-jTzLTM iFlxTk"
        src="favorites.svg"
      />
      <div>
        My Favorite Gifs
      </div>
    </div>
  </div>
  <div
    className="sc-jzJRlG cQirNv"
  >
    <div
      className="sc-bdVaJa ePiAyJ"
    >
      <div
        className="sc-bwzfXH hORBRg"
      >
        <input
          className="sc-htpNat iFKLan"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Search a gif"
          type="search"
        />
        <div
          className="sc-bxivhb gXODYk"
          id="btnSearch"
          onClick={[Function]}
        >
          <span
            className="textButton"
          >
            Search
          </span>
          <span
            className="material-icons"
          >
            search
          </span>
        </div>
      </div>
      <div
        className="sc-ifAKCX eERJos"
      />
    </div>
  </div>
  <div
    className="sc-dnqmqq ddIfsJ"
  >
    <div
      className="sc-bZQynM ilGifo"
    >
      <div
        className="sc-gzVnrw eHxNMw"
      >
        <img
          alt="loading"
          src="imgLoader.png"
        />
      </div>
      <div
        className="sc-EHOje jhQAeQ"
      >
        <img
          alt="Add to favorites"
          className="sc-htoDjs efxPqf"
          onClick={[Function]}
          src="starActive.svg"
        />
      </div>
    </div>
    <div
      className="sc-bZQynM ilGifo"
    >
      <div
        className="sc-gzVnrw eHxNMw"
      >
        <img
          alt="loading"
          src="imgLoader.png"
        />
      </div>
      <div
        className="sc-EHOje jhQAeQ"
      >
        <img
          alt="Add to favorites"
          className="sc-htoDjs efxPqf"
          onClick={[Function]}
          src="starActive.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
